agents/agents-core/src/commonMain/kotlin/ai/koog/agents/core/feature/model/featureEventTypes.kt

```diff
@@ -286,7 +282,7 @@
     val runId: String,
     val toolCallId: String?,
     val toolName: String,
-    val toolArgs: ToolArgs,
+    val toolArgs: String,
```

Comment:
Let's make it `JsonObject` instead, to indicate that it's a Json structure. Settings it just to `String` is too general, and can lead to some problems with deserialization if someone decides to set just some string here. We faced similar problems already when adding persistency support

Reply:
> could you please clarify what kind of problems do you mean?

We where setting some random strings in tests, and then in our actual logic trying to deserialize them as a valid JSON, which failed. Since tool args are always JSON, it's better to communicate this more clearly, to ensure that this parameter will always be a valid JSON and we would be able to deserialize it.